GramAddict/core/filter.py

```diff
@@ -148,3 +254,46 @@ def _has_business_category(device):
             className="android.widget.TextView",
         )
         return business_category_view.exists()
+
+    @staticmethod
+    def _is_private_account(device):
+        private = False
+        profileView = ProfileView(device)
+        try:
+            private = profileView.isPrivateAccount()
+        except Exception:
+            logger.error("Cannot find whether it is private or not")
+
+        return private
```

Comment:
There is no difference from the caller perspective whether the account is not private or the bot couldnt find the state.
Perhaps we should return None and leave it to the caller what to do?

Reply:
Fair enough. I don't think it matters if we are just talking about future but if other functions start using it - we'd want better reporting. Modified.